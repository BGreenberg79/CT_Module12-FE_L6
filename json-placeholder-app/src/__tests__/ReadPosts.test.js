import React from 'react';
import '@testing-library/jest-dom';
import PostList from '../components/PostList';
import axios from 'axios';
import { render, waitFor, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('axios')

describe('PostList Read component', ()=>{
    test('fetches posts from JSON Placeholder API upon loading', async ()=> {
        const mockResponse = {data: [{userId:1, id:1, title:'test title', body:'test body'}]};
        axios.get.mockResolvedValue(mockResponse);

        render(<PostList/>);

        await waitFor(()=> {
            expect(axios.get).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/posts')
        });

            expect(screen.getByText(/test title/i)).toBeInTheDocument();
            expect(screen.getByText(/test body/i)).toBeInTheDocument();
        });


    test('matches the snapshot', () => {
        const { asFragment } = render(<PostList />);
        expect(asFragment()).toMatchSnapshot();
    
    });
});