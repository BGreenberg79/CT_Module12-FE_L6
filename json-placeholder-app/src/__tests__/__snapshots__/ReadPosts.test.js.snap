// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList Read component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      JSON Placeholder Post List
    </h2>
    <div
      class="list-group"
    />
  </div>
</DocumentFragment>
`;
